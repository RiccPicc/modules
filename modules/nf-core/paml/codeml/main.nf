process PAML_CODEML {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/paml:4.10.7--h7b50bb2_2':
        'quay.io/biocontainers/paml:4.10.7--h7b50bb2_2' }"

    input:
    tuple val(meta), path(phy)
    path tree
    path ctl

    output:
    tuple val(meta), path("result.txt")  , emit: out_txt
    path "out.log"                       , emit: log
    path "versions.yml"                  , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    cat  ${ctl} | awk '/outfile/ { print "outfile = result.txt"; next }{ print }' | tee ${ctl} 2>&1

    codeml "${ctl}" | tee out.log 2>&1

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        paml: \$(head out.log | grep 'CODONML')
    END_VERSIONS
    """
    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch result.txt
    echo "CODONML in paml version 4.10.7, June 2023" > out.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        paml: \$(head out.log | grep 'CODONML')
    END_VERSIONS
    """
}
