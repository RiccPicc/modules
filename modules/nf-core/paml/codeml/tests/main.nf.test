// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test paml/codeml
nextflow_process {

    name "Test Process PAML_CODEML"
    script "../main.nf"
    process "PAML_CODEML"

    tag "modules"
    tag "modules_nfcore"
    tag "paml"
    tag "paml/codeml"

    test("complete config case_a") {

        when {
            process {
                """
                input[0] = [
                    [ id:'caseA' ], // meta map
                    file('https://raw.githubusercontent.com/RiccPicc/test_data/refs/heads/main/case_a.ctl', checkIfExists: true),
                    file('https://raw.githubusercontent.com/RiccPicc/test_data/refs/heads/main/case_a.phy', checkIfExists: true),
                    file('https://raw.githubusercontent.com/RiccPicc/test_data/refs/heads/main/case_a.tree', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path("out.log").exists() }
            )
        }

    }

}
