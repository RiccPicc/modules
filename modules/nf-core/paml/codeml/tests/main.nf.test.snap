{
    "stub test case_a": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "caseA"
                        },
                        "result.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "out.log:md5,da18a41fe3d2b8470b3e28e7c8a2034c"
                ],
                "2": [
                    "versions.yml:md5,b6cc6dc723c9d47d759a805d72a2f918"
                ],
                "log": [
                    "out.log:md5,da18a41fe3d2b8470b3e28e7c8a2034c"
                ],
                "out_txt": [
                    [
                        {
                            "id": "caseA"
                        },
                        "result.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b6cc6dc723c9d47d759a805d72a2f918"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-26T09:21:54.433986426"
    }
}